scripts/fuzz_opt.py

```diff
@@ -85,7 +83,7 @@ def randomize_pass_debug():
 IGNORE = '[binaryen-fuzzer-ignore]'
 
 
-def compare(x, y, comment):
+def compare(x, y, comment='comparing'):
   if x != y and x != IGNORE and y != IGNORE:
     message = ''.join([a.rstrip() + '\n' for a in difflib.unified_diff(x.split('\n'), y.split('\n'), fromfile='expected', tofile='actual')])
     raise Exception(str(comment) + ": Expected to have '%s' == '%s', diff:\n\n%s" % (
```

Comment:
You can use `.splitlines()` rather than `.split("\n")`

Reply:
Any split the diff into lines and then join it back together again? Why not just `message = difflib.unified_diff...`?

Reply:
Modifying to use `.splitlines()`, thanks.

That code is just being moved around in this PR. And originally it was copied from the emscripten test suite code. So I'm not sure why it looks that way. But testing, the output from difflib is not a list of strings, but something like a generator  that you need to do a loop to get the lines out of. [Their example shows that too](https://docs.python.org/2/library/difflib.html#difflib.unified_diff). And a newline is needed so each line is on a new line. The rstrip looks unnecessary though, I'll remove that.